### PR TITLE
Fix focus management issue when updating folder tree children

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
Fixed an issue where folder tree items could not receive focus after data was updated.

**Why it was changed:**
The change was made because consumers require the ability to update the data within the Folder Tree component.  Accessibility should not be broken by this action.

### Testing

This change was tested using:

- [X] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10350 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/test/folder-tree/FolderTreeReload.test.jsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    return null;
+  };
+
+  const reloadItems = useCallback(() => {
+    setReloaded(true);
+  }, []);
+
+  return (
+    <>
+      <button id="reload-data-button" type="button" onClick={reloadItems}> Reloa Data</button>
```
Comment (sycombs) [inline packages/terra-framework-docs/src/terra-dev-site/test/folder-tree/FolderTreeReload.test.jsx:145]: tiny nit:
```suggestion
      <button id="reload-data-button" type="button" onClick={reloadItems}> Reload Data</button>
```

**Review by sycombs — COMMENTED** [review]
Code looks good to me. I noticed some unexpected behavior with the reload example though where prior to clicking the "Reload Data" button, the expand all button would expand all items as expected, but after clicking "Reload Data" it would no longer expand the items. Individually row expand/collapse works fine. Most likely the expanded items state needs to be updated when the data is reloaded as well.

File: packages/terra-framework-docs/src/terra-dev-site/test/folder-tree/FolderTreeReload.test.jsx (hunk 1 above)
Comment (cm9361) [reply packages/terra-framework-docs/src/terra-dev-site/test/folder-tree/FolderTreeReload.test.jsx:145]: Good catch :)

**Comment by cm9361** [comment]
> Code looks good to me. I noticed some unexpected behavior with the reload example though where prior to clicking the "Reload Data" button, the expand all button would expand all items as expected, but after clicking "Reload Data" it would no longer expand the items. Individually row expand/collapse works fine. Most likely the expanded items state needs to be updated when the data is reloaded as well.

I am pretty sure I got overzealous on the useCallback usage.  I'll fix it.

**Comment by cm9361** [comment]
@sycombs - I fixed the test page issue.

**Review by sycombs — APPROVED** [review]

**Review by kenk2 — APPROVED** [review]

**Pull request merged**